examples/pytorch/correct_and_smooth/main.py

```diff
@@ -0,0 +1,183 @@
+import argparse
+import copy
+import os
+import torch
+import torch.nn.functional as F
+import torch.optim as optim
+import dgl
+from ogb.nodeproppred import DglNodePropPredDataset, Evaluator
+from model import MLP, MLPLinear, GAT, CorrectAndSmooth
+
+
+def evaluate(y_pred, y_true, idx, evaluator):
+    return evaluator.eval({
+        'y_true': y_true[idx],
+        'y_pred': y_pred[idx]
+    })['acc']
+
+
+def main():
+    # check cuda
+    device = f'cuda:{args.gpu}' if torch.cuda.is_available() and args.gpu >= 0 else 'cpu'
+    # load data
+    dataset = DglNodePropPredDataset(name=args.dataset)
+    evaluator = Evaluator(name=args.dataset)
+
+    split_idx = dataset.get_idx_split()
+    g, labels = dataset[0] # graph: DGLGraph object, label: torch tensor of shape (num_nodes, num_tasks)
+    
+    if args.dataset == 'ogbn-arxiv':
+        if args.model == 'gat':
+            srcs, dsts = g.all_edges()
+            g.add_edges(dsts, srcs)
+            g = g.add_self_loop()
+        else:
+            g = dgl.to_bidirected(g, copy_ndata=True)
```

Comment:
Can you try replacing `g = dgl.to_bidirected(g, copy_ndata=True)` with `g = dgl.add_reverse_edge(g, copy_ndata=True)`. `to_bidirected` also removes duplicate edges in addition to `add_reverse_edge`. You can then call `g = g.remove_self_loop().add_self_loop()` for the case of `gat`.